File: Class/class05_230122/square-of-a-sorted-arrray.py
def sorted_square(nums):
    n=len(nums)
    j=n-1
    for i in range(n):
        nums[i]=nums[i]*nums[i]

    while(j>0):
        print(nums)
        if(nums[0]>nums[j]):
            tmp=nums[0]
            nums[0]=nums[j]
            nums[j]=tmp
        j-=1

    return nums


nums=[-4,-1,0,3,10]
print(sorted_square(nums))


=============================================================

class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        n=len(nums)

#         #Approach1
#         j=n-1
#         for i in range(n):
#             nums[i]=nums[i]*nums[i]
#         i=0
#         while(j>i):
#             print(nums)
#             if(nums[i]>nums[j]):
#                 nums[i],nums[j] = nums[j], nums[i]
#                 if nums[i]<=nums[j-1]:
#                     i+=1
#             j-=1

#         return nums

        #Approach2
        low=0
        high=n-1
        res=[None]*n
        idx=n-1
        while low<high:
            lowSq=nums[low]*nums[low]
            highSq=nums[high]*nums[high]
            if lowSq<highSq:
                print(idx)
                res[idx]=highSq
                high-=1
                idx-=1
            else:
                print(idx)
                res[idx]=lowSq
                low+=1
                idx-=1
        return res

File: Class/class05_230122/square-of-a-sorted-arrray.java
//============================================================
    //Approach2, JAVA Solution
/*
public int[] sortedSquares(int[] nums)
{
    int low =0,high=nums.length-1;
    int[] res = new int[nums.length];
    int idx = nums.length-1;
    while(low<=high)
    {
        int lowSq = nums[low]*nums[low];
        int highSq = nums[high]*nums[high];
        if(lowSq<highSq)
        {
            res[idx--]=highSq;
            high--;
        }
        else
        {
            res[idx--] = lowSq;
            low++;
        }
    }
    return res;
}
*/

//=================================================================
// Java program to Sort square of
// the numbers of the array


import java.util.*;
import java.io.*;

//class GFG{

// Function to sort an square array
public static void sortSquares(int arr[])
{
	int n = arr.length, left = 0,
		right = n - 1;

	int result[] = new int[n];

	for(int index = n - 1; index >= 0; index--)
	{
		if (Math.abs(arr[left]) > arr[right])
		{
			result[index] = arr[left] * arr[left];
			left++;
		}
		else
		{
			result[index] = arr[right] * arr[right];
			right--;
		}
	}
	for(int i = 0; i < n; i++)
		arr[i] = result[i];
}

// Driver code
public static void main(String[] args)
{
	int arr[] = { -6, -3, -1, 2, 4, 5 };
	int n = arr.length;

	System.out.println("Before sort ");
	for(int i = 0; i < n; i++)
		System.out.print(arr[i] + " ");

	sortSquares(arr);
	System.out.println("");
	System.out.println("After Sort ");
	for(int i = 0; i < n; i++)
		System.out.print(arr[i] + " ");
}
//}

// This code is contributed by jinalparmar2382
